lib/viking/record/association.js

```diff
@@ -84,19 +83,16 @@ export default class Association extends EventBus {
         return this.load()
     }
     
-    saveAttributes (options) {
-        if (this.target) {
-            const attributes = this.target.saveAttributes(options);
-            if (this.target.isPersisted() && Object.keys(attributes).length > 0) {
-                attributes[this.target.constructor.primaryKey] = this.target.primaryKey()
-            }
-            return attributes;
-        } else if (this.removedRecord) {
-            const attributes = {}
-            attributes[this.removedRecord.constructor.primaryKey] = this.removedRecord.primaryKey()
-            attributes._destroyed = true
-            return attributes
+    autosave() {
+        return (this.target && this.target.hasChanged())
```

Comment:
```suggestion
        return (this.target && this.target.needsSaved())
```
The target could have sub resources that need saving